x/vesting/keeper/integration_test.go

```diff
@@ -55,75 +63,153 @@ var _ = Describe("Clawback Vesting Accounts", Ordered, func() {
-		clawbackAccount = types.NewClawbackVestingAccount(
-			baseAccount,
-			funder,
-			vestingAmtTotal,
-			vestingStart,
-			lockupPeriods,
-			vestingPeriods,
-		)
-		err := testutil.FundAccount(s.ctx, s.app.BankKeeper, addr, vestingAmtTotal)
-		s.Require().NoError(err)
-		acc := s.app.AccountKeeper.NewAccount(s.ctx, clawbackAccount)
-		s.app.AccountKeeper.SetAccount(s.ctx, acc)
+		// Initialize all test accounts
+		for i, account := range testAccounts {
+			// Create and fund periodic vesting account
+			vestingStart := s.ctx.BlockTime()
+			baseAccount := authtypes.NewBaseAccountWithAddress(account.address)
+			clawbackAccount = types.NewClawbackVestingAccount(
+				baseAccount,
+				funder,
+				vestingAmtTotal,
+				vestingStart,
+				lockupPeriods,
+				vestingPeriods,
+			)
 
-		// Check if all tokens are unvested at vestingStart
-		unvested = clawbackAccount.GetUnvestedOnly(s.ctx.BlockTime())
-		vested = clawbackAccount.GetVestedOnly(s.ctx.BlockTime())
-		s.Require().Equal(vestingAmtTotal, unvested)
-		s.Require().True(vested.IsZero())
+			err := testutil.FundAccount(s.ctx, s.app.BankKeeper, account.address, vestingAmtTotal)
+			s.Require().NoError(err)
+			acc := s.app.AccountKeeper.NewAccount(s.ctx, clawbackAccount)
+			s.app.AccountKeeper.SetAccount(s.ctx, acc)
+
+			// Check if all tokens are unvested at vestingStart
+			unvested = clawbackAccount.GetUnvestedOnly(s.ctx.BlockTime())
+			vested = clawbackAccount.GetVestedOnly(s.ctx.BlockTime())
+			s.Require().Equal(vestingAmtTotal, unvested)
+			s.Require().True(vested.IsZero())
+
+			// Grant gas stipend to cover EVM fees
+			err = testutil.FundAccount(s.ctx, s.app.BankKeeper, clawbackAccount.GetAddress(), accountGasCoverage)
+			s.Require().NoError(err)
+			granteeBalance := s.app.BankKeeper.GetBalance(s.ctx, account.address, stakeDenom)
+			s.Require().Equal(granteeBalance, accountGasCoverage[0].Add(vestingAmtTotal[0]))
+
+			// Update testAccounts clawbackAccount reference
+			testAccounts[i].clawbackAccount = clawbackAccount
+		}
 	})
 
 	Context("before first vesting period", func() {
+		BeforeEach(func() {
+			// Add a commit to instantiate blocks
+			s.Commit()
+
+			// Ensure no tokens are vested
+			vested := clawbackAccount.GetVestedOnly(s.ctx.BlockTime())
+			unlocked := clawbackAccount.GetUnlockedOnly(s.ctx.BlockTime())
+			zeroCoins := sdk.NewCoins(sdk.NewCoin(stakeDenom, sdk.ZeroInt()))
+			s.Require().Equal(zeroCoins, vested)
+			s.Require().Equal(zeroCoins, unlocked)
+		})
+
 		It("cannot delegate tokens", func() {
-			err := delegate(clawbackAccount, 100)
+			err := delegate(clawbackAccount, math.NewInt(100))
 			Expect(err).ToNot(BeNil())
 		})
 
-		It("cannot transfer tokens", func() {
+		It("can transfer spendable tokens", func() {
+			account := testAccounts[0]
+			// Fund account with new spendable tokens
+			testutil.FundAccount(s.ctx, s.app.BankKeeper, account.address, unvested)
+
 			err := s.app.BankKeeper.SendCoins(
```

Comment:
Fix lint issue, check for error
```suggestion
			err := testutil.FundAccount(s.ctx, s.app.BankKeeper, account.address, unvested)
			s.NoError(err)

			err = s.app.BankKeeper.SendCoins(
```